### PR TITLE
SQLite support

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR replace MongoDB with SQLite. SQLite is serverless, self-contained and doesn't need any configuration. It should be a good choice for our first beta version.

 The default DB file path is `dist/data/zk-groups.db`. [`sqlitebrowser`](https://sqlitebrowser.org/dl/) is a good and open-source SQLite browser.

## Relate Issue

#38 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Review by Insun35 — APPROVED** [review]

**Review by jdhyun09 — APPROVED** [review]

File: apps/api/src/app/accounts/account.service.ts [hunk 1]
```diff
@@ -1,49 +1,46 @@
...
-            | FindOptionsWhere<AccountModel>[]
-    ): Promise<AccountModel> {
+        payload: FindOptionsWhere<Account> | FindOptionsWhere<Account>[]
+    ): Promise<Account> {
         return this.accountRepository.findOneBy(payload)
     }
 
     public async update(
-        id: ObjectID,
+        id: number,
```
Comment (Insun35) [inline apps/api/src/app/accounts/account.service.ts:37]: There is an error that id is not unique.
SQLITE_CONSTRAINT: UNIQUE constraint failed: invites.groupId

**Review by Insun35 — CHANGES_REQUESTED** [review]
There is an error that id is not unique.
And also it does not return the group data.

File: apps/api/src/app/invites/entities/invite.entity.ts [hunk 2]
```diff
@@ -1,19 +1,18 @@
 /* istanbul ignore file */
-import { ObjectId } from "mongodb"
 import {
     Column,
     Entity,
     Index,
     JoinColumn,
-    ObjectIdColumn,
-    OneToOne
+    OneToOne,
```
Comment (jdhyun09) [inline apps/api/src/app/invites/entities/invite.entity.ts:7]: I found the reason why the invite code is only generated once. `@OnetoOne` can have relation with only one invite code per group.

I found `@ManytoOne` to be the solution here.

Please tell me if I'm wrong!

File: apps/api/src/app/accounts/account.service.ts (hunk 1 above)
Comment (cedoor) [reply apps/api/src/app/accounts/account.service.ts:37]: Hey, are you running `yarn start api`?

File: apps/api/src/app/invites/entities/invite.entity.ts (hunk 2 above)
Comment (cedoor) [reply apps/api/src/app/invites/entities/invite.entity.ts:7]: I think OneToOne is correct for invites, but maybe there should by a OneToMany `invites` field in the group entity.

Comment (cedoor) [reply apps/api/src/app/invites/entities/invite.entity.ts:7]: Let me check! But you're right I guess

Comment (cedoor) [reply apps/api/src/app/invites/entities/invite.entity.ts:7]: Good catch! I'm going to fix it

**Comment by cedoor** [comment]
> There is an error that id is not unique. And also it does not return the group data.

It should work now, you can try this code to get group data!

```ts
const invite = await this.inviteRepository.findOne({
            where: {
                code: inviteCode
            },
            relations: ["group"]
        })
```

**Review by jdhyun09 — APPROVED** [review]

**Review by Insun35 — APPROVED** [review]

**Pull request merged**